### DISCUSSION TITLE
Simple solution for using a flow to store images as DirectusFile in storage.

### DISCUSSION THREAD

**Discussion by SvenBudak** [directus/directus#21018] [Feature Requests]
### Summary

Allows for crawling many images in one flow and saving them as DirectusFile.

### Basic Example

```ts
// Example of a new flow step for image crawling
Flow.create('Image Crawler')
    .setSource('URL List')
    .crawlImages()
    .storeAsDirectusFiles()
    .execute();
```

### Motivation

The motivation for this feature stems from the need to efficiently handle large quantities of image data within Directus. Current methods involve manual uploading or less optimal programmatic approaches, which are time-consuming and not scalable. This feature supports use cases like bulk image uploads for e-commerce platforms, digital archives, and content management systems, aiming to simplify and automate the process, thereby saving time and reducing errors.

### Overview

The proposed feature extends the functionality of Directus Flows to allow for efficient processing and storage of a large number of images. The aim is to implement an automated solution enabling users to crawl images from various sources and directly save them as DirectusFiles in the Directus database.

### Technical Implementation

1. **Extension of Flow Component:** The existing Flow component in Directus will be expanded to include a new function that processes image URLs. This function will be integrated as a new step within a flow.
2. **Crawler Functionality:** Within the flow, a crawler will be implemented that receives the URLs of the images, downloads the images, and temporarily stores them.
3. **Integration with Directus File Storage:** After downloading the images, an automatic integration into Directus File Storage takes place. Each downloaded image is created as a new DirectusFile object, taking into account relevant metadata such as file name, size, and type.
4. **User Interface for Configuration:** A user-friendly interface will be provided where users can input URLs or upload a list of URLs. Additionally, users can set preferences regarding file size limits and file types.
5. **Error Handling and Logging:** A robust error handling system will be implemented to deal with invalid URLs or download errors. A logging system will record successful and failed operations for monitoring and troubleshooting.

### Examples and Use Cases

A typical use case might be the bulk import of product images for an online store. Instead of manually downloading and uploading each image to Directus, a user could insert a list of product image URLs into the flow, automating the process.

### Integration and Compatibility

The feature will be developed to seamlessly interact with existing components of Directus, particularly the file management system. Care will be taken to ensure that the feature does not impair Directus's performance and remains compatible with future versions.

### Must Have:

- Efficient crawling of images from provided URLs.
- Automatic storage of crawled images as DirectusFiles.
- Integration with the Directus file management system.
- User-friendly interface for inputting image URLs.

### Should Have:

- Customizable settings for file types and size limits.
- Robust error handling and logging for failed downloads.

### Could Have:

- Batch processing capabilities for handling very large sets of images.
- Advanced filtering options for image selection.

### Won't Have:

- Real-time image editing or manipulation features.
- Integration with external image hosting platforms.

### Drawbacks

The main drawbacks include the potential increase in complexity within the Directus system and the resources required for the development and maintenance of this feature. Additionally, integrating this new functionality might require careful consideration to ensure it does not disrupt existing features or workflows within Directus.

### Alternatives

An alternative considered is enhancing the existing file import feature in Directus to better handle bulk operations. However, this approach lacks the automation and scalability offered by the proposed image crawling functionality.

### Adoption Strategy

The feature will be designed as an optional extension to the existing Directus Flows, ensuring it does not introduce breaking changes. Existing Directus users can adopt the feature by updating to the latest version and optionally incorporating the new flow into their processes.

**Comment by ched-dev** [comment]
This could be achieved with some custom operations, but it probably isn't something the 80% of users would need.

**Reply by domdus** [reply]
There is actually a custom operation available for images:
https://github.com/samechikson/directus-extension-file-import-operation

Using this you could create a flow loop upfront to process a batch of images / image urls.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
